### PR TITLE
sync internal node ready condition to v3 node condition

### PR DESCRIPTION
**Pull request by kinarashah**
**Issue:**
When powered down, node shows Unavailable from Cluster explorer view but shows up as Active from cluster management view. 

**Solution**
Norman embeds internalNodeStatus.Conditions to node.status.Conditions which causes the ready condition to show up correctly on v3 calls. Syncing it to v3 node.status.conditions for steve v1 node calls to get the same effect. 

https://github.com/rancher/rancher/issues/35646

### CODE REVIEW

File: pkg/controllers/managementuser/nodesyncer/nodessyncer.go [hunk 1]
```diff
@@ -593,11 +595,11 @@ func objectsAreEqual(existing *v3.Node, toUpdate *v3.Node) bool {
 	rolesEqual := toUpdateToCompare.Spec.Worker == existingToCompare.Spec.Worker && toUpdateToCompare.Spec.Etcd == existingToCompare.Spec.Etcd &&
 		toUpdateToCompare.Spec.ControlPlane == existingToCompare.Spec.ControlPlane
 
-	retVal := statusEqual && specEqual && nodeNameEqual && labelsEqual && annotationsEqual && requestsEqual && limitsEqual && rolesEqual
+	retVal := statusEqual && conditionsEqual && specEqual && nodeNameEqual && labelsEqual && annotationsEqual && requestsEqual && limitsEqual && rolesEqual
 	if !retVal {
-		logrus.Debugf("ObjectsAreEqualResults for %s: statusEqual: %t specEqual: %t"+
+		logrus.Debugf("ObjectsAreEqualResults for %s: statusEqual: %t conditionsEqual %t specEqual: %t"+
```
Comment (snasovich) [inline pkg/controllers/managementuser/nodesyncer/nodessyncer.go:600]: Missing `:` after `conditionsEqual`

File: pkg/controllers/managementuser/nodesyncer/nodessyncer.go [hunk 2]
```diff
@@ -676,6 +678,24 @@ func cleanStatus(machine *v3.Node) {
...
+				LastTransitionTime: condition.LastTransitionTime.String(),
+				LastUpdateTime:     condition.LastHeartbeatTime.String(),
+				Reason:             condition.Reason,
+				Message:            condition.Message,
+			}
+			var exists bool
+			for i, cond := range machine.Status.Conditions {
+				if cond.Type == "Ready" {
+					exists = true
+					machine.Status.Conditions[i] = readyCondition
```
Comment (snasovich) [inline pkg/controllers/managementuser/nodesyncer/nodessyncer.go:693]: Can we break out of loop here as we've already found `Ready` condition? Not sure if it's theoretically possible to have multiple conditions with the same type.

**Review by snasovich — CHANGES_REQUESTED** [review]
Couple minor requests/questions.

However, on the overall approach - we're copying Ready condition from `internalNodeStatus` up to root status for `v3.Node` objects. Where does the syncing of conditions to `v1.Node` happens? In here I only see `v1.Node` updated in `updateLabels` and `updateAnnotations` none of which would be triggered if conditions on the status is the only thing that has changed. Does v3 -> v1 syncing happens elsewhere?

**Comment by kinarashah** [comment]
@snasovich Thanks for the review, will update review comments. 
About syncing to v1 node, I don't understand why we need to sync it to v1 from v3 node? Here v3 is rancher's management node and v1 is core k8s downstream node. From cluster management view, UI relies on the v3 management node's conditions so we sync the ready condition from k8s core v1 node to v3 management node. Users have the option set labels and annotations on v3 management node, so we need to sync them to k8s core v1 downstream node.

**Comment by snasovich** [comment]
@kinarashah , thank you for the clarifications. 👍 
There was a misunderstanding on my end. I figured UI shows nodes based on the response from `v1/management.cattle.io.nodes` API call and automatically assumed we have `v1` version of Node object somewhere in Rancher CRDs. However that API call returns Node objects from `management.cattle.io/v3` so we're good. All these `v1`s / `v3`s are a bit confusing sometimes.

File: pkg/controllers/managementuser/nodesyncer/nodessyncer.go (hunk 2 above)
Comment (kinarashah) [reply pkg/controllers/managementuser/nodesyncer/nodessyncer.go:693]: done

**Review by snasovich — APPROVED** [review]

**Pull request merged**